Trakt/Api/TraktController.cs

```diff
@@ -6,152 +6,160 @@
-    [ProducesResponseType(StatusCodes.Status200OK)]
-    public ActionResult<object> TraktPollAuthorizationStatus([FromRoute] string userId)
-    {
-        _logger.LogInformation("TraktPollAuthorizationStatus request received");
-        var traktUser = UserHelper.GetTraktUser(userId);
-        bool isAuthorized = traktUser.AccessToken != null && traktUser.RefreshToken != null;
+            _logger.LogInformation("TraktDeviceAuthorization request received");
+
+            // Create a user if we don't have one yet - TODO there should be an endpoint for this that creates a default user
+            var traktUser = UserHelper.GetTraktUser(userId);
+            if (traktUser == null)
+            {
+                Plugin.Instance.PluginConfiguration.AddUser(userId);
+                traktUser = UserHelper.GetTraktUser(userId);
+                Plugin.Instance.SaveConfiguration();
+            }
+
+            string userCode = await _traktApi.AuthorizeDevice(traktUser).ConfigureAwait(false);
+
+            return new
+            {
+                userCode
+            };
+        }
 
-        if (Plugin.Instance.PollingTasks.TryGetValue(userId, out var task))
+        /// <summary>
+        /// Poll the trakt.tv device authorization status.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <response code="200">Polling successful.</response>
+        /// <returns>A value indicating whether the authorization code was connected to a trakt.tv account.</returns>
+        [HttpGet("Users/{userId}/PollAuthorizationStatus")]
+        [Authorize(Policy = "RequiresElevation")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<object> TraktPollAuthorizationStatus([FromRoute] string userId)
         {
-            isAuthorized = task.Result;
-            Plugin.Instance.PollingTasks.Remove(userId);
+            _logger.LogInformation("TraktPollAuthorizationStatus request received");
+            var traktUser = UserHelper.GetTraktUser(userId);
+            bool isAuthorized = traktUser.AccessToken != null && traktUser.RefreshToken != null;
+
+            if (Plugin.Instance.PollingTasks.TryGetValue(userId, out var task))
+            {
+                isAuthorized = task.Result;
+                Plugin.Instance.PollingTasks.Remove(userId);
+            }
+
+            return new
+            {
+                isAuthorized
+            };
         }
 
-        return new
+        /// <summary>
+        /// Rate an item.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <param name="itemId">The item id.</param>
+        /// <param name="rating">Rating between 1 - 10 (0 = unrate).</param>
+        /// <response code="200">Item rated successfully.</response>
+        /// <returns>A <see cref="TraktSyncResponse"/>.</returns>
+        [HttpPost("Users/{userId}/Items/{itemId}/Rate")]
+        [Authorize(Policy = "DefaultAuthorization")]
```

Comment:
This isn't needed as it's already being applied at the controller level
`RequiresElevation` should stay though